### ISSUE TITLE
Unhandled rejection Error: ENOENT on npm update -g

### ISSUE THREAD

**Issue by Erikmitk** [npm/npm#17800]
#### I'm opening this issue because:
  - [x] npm is doing something I don't understand.

#### What's going wrong?

npm fails to update (keybase apparently) and tells me to report this error to you.

```
Unhandled rejection Error: ENOENT: no such file or directory, stat '/var/folders/pj/v6klljn57w73qrfv1nfq_v0w0000gn/T/keybase_install_kgfT3DoXJHNX__U6J3_fxQ/keybase-0.7.2.tgz'

npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/emk/.npm/_logs/2017-07-15T20_24_28_439Z-debug.log
```

#### How can the CLI team reproduce the problem?

The debug.log is here: https://gist.github.com/Erikmitk/7031992ac4682e9ed05fc1feee16d777

### supporting information:

 - `npm -v` prints: 5.3.0
 - `node -v` prints: v8.1.4
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: macOS

**Comment by pinkahd** [comment]
I'm having the same issue, I believe the issues is when installing a package globally and the package folder was removed later on.

### How can the CLI team reproduce the problem?

To replicated the issues install a global package from a local repository and then delete the repository then run `npm update -g`. This seems to replicate the exact behavior. 

Debug log: https://gist.github.com/pinkahd/b4448e7e9dee4472f5795c78dd5e30af

### Supporting information:

- npm -v prints: 5.3.0
- node -v prints: v6.10.3
- npm config get registry prints: https://registry.npmjs.org/
- Windows,  OS X/macOS, or Linux?: macOS

**Comment by ActionScripted** [comment]
Same issue here. For me it was after updating node/npm from Homebrew. Full output:

```
npm update -g
Unhandled rejection Error: ENOENT: no such file or directory, stat '/Users/brew/Library/Caches/Homebrew/node--npm-5.3.0.tgz'

npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mthompson/.npm/_logs/2017-07-20T15_23_10_534Z-debug.log
```

### How can the CLI team reproduce the problem?

Debug log: https://gist.github.com/ActionScripted/934928427d91208cbe2c7de87f88d1dc

### Supporting information:
- `npm -v` prints: 5.3.0
- `node -v` prints: v8.2.0
- `npm config get registry` prints: https://registry.npmjs.org/
- Windows, OS X/macOS, or Linux?: macOS

### Things I've tried:
- Uninstalling all global packages and then (re)installing them
- Uninstalling node and (re)install node

### Total speculation:
It looks like it tries to find the `node--npm-5.3.0.tgz` relative to where `npm update -g` is run on my system. If I change directories to something like `/Users/myuser/somepath/` the `Unhandled rejection Error: ENOENT: no such file or directory` will point to `/Users/myuser/somepath/Users/brew/Library/Caches/Homebrew/node--npm-5.3.0.tgz`

Not sure what that means or if it's even a problem but it's something I've noticed.

**Comment by ramyareye** [comment]
Same for me! I'm using Mac and brew! and I know I did something wrong but don't know where and when and how to fix it ! 

`Ramyar:Desktop Ramyar$ npm outdated -g --depth=0`
`Unhandled rejection Error: ENOENT: no such file or directory, open '/Users/Users/Ramyar/Documents/React/parse-server/package.json'`

any help ? :-)

**Comment by Erikmitk** [comment]
This appears to be fixed with npm 5.4.0. I cannot reproduce this error on my end anymore.
